bench/db_bench.ml

```diff
@@ -219,188 +198,127 @@ module Index = struct
-    Array.sort (fun a b -> String.compare (fst a) (fst b)) random;
-    get_wtxn root flags >>| fun (rw, env) ->
-    print_results (write rw) nb_entries;
-    closedir env
-
-  let write_sync () =
-    get_wtxn root [ Lmdb.NoRdAhead ] >>| fun (rw, env) ->
-    let write (txn, ddb) env ls () =
-      Array.iter
-        (fun (k, v) ->
-          fail_on_error (fun () ->
-              Lmdb.put_string txn ddb k v >>= fun () -> sync env))
-        ls
-    in
-    print_results (write rw env random) nb_entries;
-    closedir env
-
-  let overwrite rw = print_results (write rw) nb_entries
-
-  let read_random r = print_results (read r) nb_entries
-
-  (*use a new db, created without the flag Lmdb.NoRdAhead*)
-  let read_seq () =
-    let rw, env =
-      R.get_ok
-        ( get_wtxn root [ Lmdb.NoSync; Lmdb.NoMetaSync ] >>| fun (rw, env) ->
-          let () = write rw () in
-          (rw, env) )
-    in
-    let read (txn, ddb) () =
-      opencursor txn ddb >>= fun cursor ->
-      cursor_first cursor >>= fun () ->
-      cursor_iter
-        ~f:(fun (k, v) ->
-          ignore (Bigstring.to_string k);
-          ignore (Bigstring.to_string v);
-          Ok ())
-        cursor
-      >>| fun () -> cursor_close cursor
-    in
-    let aux_read r () = fail_on_error (read r) in
-    print_results (aux_read rw) nb_entries;
-    closedir env
-
-  let close env = closedir env
-end
-
 let init () =
   Common.report ();
   Index.init ();
-  Lmdb.cleanup ();
   Log.app (fun l -> l "Keys: %d bytes each." key_size);
   Log.app (fun l -> l "Values: %d bytes each." value_size);
   Log.app (fun l -> l "Entries: %d." nb_entries);
   Log.app (fun l -> l "Log size: %d." log_size);
   populate ()
 
-let index_main () =
-  Log.app (fun l -> l "\n");
-  Log.app (fun l -> l "Fill in increasing order of keys");
-  Index.write_seq ();
-  Log.app (fun l -> l "\n");
-  Log.app (fun l -> l "Fill in increasing order of hashes");
-  Index.write_seq_hash ();
-  Log.app (fun l -> l "\n");
-  Log.app (fun l -> l "Fill in decreasing order of hashes");
-  Index.write_rev_seq_hash ();
-  populate ();
-  Log.app (fun l -> l "\n");
-  Log.app (fun l -> l "Fill in random order and sync after each write");
-  Index.write_sync ();
+open Cmdliner
+
+let run input =
+  init ();
   Log.app (fun l -> l "\n");
   Log.app (fun l -> l "Fill in random order");
   let rw = Index.write_random () in
-  Log.app (fun l -> l "\n");
-  Log.app (fun l -> l "Read in random order ");
-  Index.read_random rw;
-  Log.app (fun l -> l "\n");
-  Log.app (fun l ->
-      l
-        "Read in sequential order (increasing order of hashes for index, \
-         increasing order of keys for lmdb)");
-  Index.read_seq rw;
-  Log.app (fun l -> l "\n");
-  Log.app (fun l -> l "Overwrite");
-  Index.overwrite rw;
+  let () =
+    match input with
+    | `Find `RW | `IndexAll | `All | `Minimal ->
+        let () = Log.app (fun l -> l "\n RW Read in random order") in
+        Index.read_random rw
+    | _ -> ()
```

Comment:
How about extracting this logic to a function? It's used in quite a lot of places. Maybe something like:
```ocaml
val may_run : 
    ~bench:(unit -> unit) ->
    ~name:string ->
    ~triggers:bench_id list ->
    ~input:bench_id -> 
    unit
```
Then maybe you could store all of the benchmarks in a list and `List.iter` over it, rather than repeatedly matching on the input.